src/main/java/net/minecraftforge/fml/common/eventhandler/Event.java

```diff
@@ -90,7 +91,10 @@ public void setCanceled(boolean cancel)
     {
         if (!isCancelable())
         {
-            throw new IllegalArgumentException("Attempted to cancel a non cancellable event");
+            throw new UnsupportedOperationException(
+                "Attempted to modify the unmodifiable canceled state of a non-cancelable event of type: "
```

Comment:
I was trying to capture the fact that the error would be thrown even if you pass `true` to the method. I concede, however, that my wording is not optimal.

Would say, `"Attempted to call Event#setCanceled() on a non-cancelable event of type: <type goes here>"` be better?